pkg/apps/watch-appstream.py

```diff
@@ -0,0 +1,253 @@
+import os
+import sys
+import time
+import traceback
+
+import gzip
+import xml.etree.ElementTree as ET
+import json
+
+lang = os.environ.get("LANGUAGE")
+
+def attr_lang(elt):
+    return elt.attrib.get('{http://www.w3.org/XML/1998/namespace}lang')
+
+def element(xml, tag):
+    if lang:
+        for elt in xml.iter(tag):
+            if attr_lang(elt) == lang:
+                return elt
+    return xml.find(tag)
+
+def element_value(xml, tag):
+    elt = element(xml, tag)
+    return elt.text if elt is not None else None
+
+def convert_description(xml, use_lang = True):
+    if xml is None:
+        return None
+
+    want_lang = lang if use_lang else None
+
+    # Only the following constructs are allowed, and they all appear
+    # at the top level:
+    #
+    # <p>text</p>
+    # <ul><li>text</li>...</ul>
+    # <ol><li>text</li>...</ol>
+
+    # A description can have 'lang' attributes both on the actual
+    # <description> element and on the contained <p>, <ul>, and <ol>
+    # elements, but probably not on the <li> elements.
+
+    def text(xml):
+        return " ".join(xml.itertext())
+
+    res = [ ]
+    for c in xml:
+        if attr_lang(c) != want_lang:
+            continue
```

Comment:
If `want_lang` is `None`, that would skip the entire loop. It seems this is actually intended (there's a fallback below), but it's confusing to read here. Maybe add a comment?

Reply:
`attr_lang(c)` can also be None, no?

Reply:
Yes. I think the logic is okay as-is, it's just not very obvious when reading the code. Hence the suggestion to explain what this is doing in a comment.

Reply:
Hmm, I don't see this as particularly hairy.  I would rather rewrite the code to be clearer.

The loop looks at all elements and ignores those for a wrong lang.  Would it help to invert the condition?  Or use a comprehension?  A filter?